src/astro/databases/base.py

```diff
@@ -201,15 +201,44 @@ def append_table(
         self,
         source_table: Table,
         target_table: Table,
-        if_conflicts: AppendConflictStrategy = "exception",
+        source_tables_cols: Optional[List[str]] = None,
+        target_tables_cols: Optional[List[str]] = None,
```

Comment:
The table definition has a property `columns`:
https://github.com/astronomer/astro-sdk/blob/main/src/astro/sql/table.py#L45

Why aren't we using it in these two lines?

Reply:
we need a separate set of cols since the user can choose a subset of cols as opposed to all the cols specified in the Table object, which is a user-defined construct and altering is not something we should do.